src/hooks/usePoolsHandler.tsx

```diff
@@ -1,38 +1,84 @@
-import { useEffect, useState } from "react"
+import { useEffect, useState } from 'react'
+import { callApi } from '../utils/graphQLApi'
+import { notifyError } from './useNotification'
 
-type PoolType = { asset: string; totalSupply: string; supplyApy: string; totalBorrowed: string; borrowApyVariable: string; borrowApyStable: string }
+type PoolType = {
+  asset: string
+  totalSupply: string
+  supplyApy: string
+  totalBorrowed: string
+  borrowApyVariable: string
+  borrowApyStable: string
+}
 
 export const usePoolsHandler = () => {
-  const [poolData, setPoolData] = useState<PoolType[]>()
+  const [poolData, setPoolData] = useState<PoolType[]>([])
 
   useEffect(() => {
-    setPoolData([
-      {
-        asset: "Binance",
-        totalSupply: "20,223,182,626",
-        supplyApy: "895",
-        totalBorrowed: "20,960,370",
-        borrowApyVariable: "1683",
-        borrowApyStable: "394",
-      },
-      {
-        asset: "Coinbase Exchange",
-        totalSupply: "2,815,007,914",
-        supplyApy: "945",
-        totalBorrowed: "1,756,438",
-        borrowApyVariable: "1274",
-        borrowApyStable: "5196",
-      },
-      {
-        asset: "Gate.io",
-        totalSupply: "1,652,717,489",
-        supplyApy: "549",
-        totalBorrowed: "2,658,869",
-        borrowApyVariable: "2565",
-        borrowApyStable: "1448",
-      },
-    ])
-  }, [])
+    if (poolData?.length === 0) {
+      let query = JSON.stringify({
+        query: `
+    {
+      poolCreateds {
+        id
+        pool
+        cfmm
+        protocolId
+        protocol
+        count
+      }
+    }
+    `,
+      })
+      callApi(query)
+        .then((res) => {
```

Comment:
Just one comment from me, asyc/await seems to be much more readable than many nested then statements. We should put this as a standard.

ie. let res = await callApi(query)
thenFunc(res)

Reply:
Done